driver/src/orderbook/event_based_orderbook/orderbook.rs

```diff
@@ -0,0 +1,110 @@
+// Ethereum events (logs) can be both created and removed. Removals happen if the chain reorganizes
+// and ends up not including block that was previously thought to be part of the chain.
+// However, the orderbook state (`State`) cannot remove events. To support this, we keep an ordered
```

Comment:
Why can the orderbook state not remove events? What information is not reflected in the event data?

I'd assume there is a way to make sure events are always applied and reverted "in order".

Reply:
Because not all operations are reversible. The simplest case is the deletion of an order. Once it has been deleted we can't undo the delete because we do not know what the original order was.